### PR TITLE
examples/spark: resolve failures on Cray XC40

### PR DESCRIPTION
**Pull request by heasterday**
Addresses #978

We can get fancier for setting JAVA_HOME but this location appears to be consistent on Debian.

Marking as draft pending more tests.

### CODE REVIEW

**Comment by heasterday** [comment]
Passes tests on local dev machine, xc40, and CTS-1.

File: examples/spark/test.bats [hunk 1]
```diff
@@ -31,22 +31,16 @@ setup () {
...
     spark_log=/tmp/sparklog
     if [[ $ch_multinode ]]; then
-        # Use the last non-loopback IP address. This is a barely educated
-        # guess and shouldn't be relied on for real code, but hopefully it
-        # works for testing.
-        master_ip=$(  ip -o -f inet addr show \
-                    | grep -F 'scope global' \
-                    | tail -1 \
-                    | sed -r 's/^.+inet ([0-9.]+).+/\1/')
+        master_host="$(hostname)"
```
Comment (reidpr) [inline examples/spark/test.bats:34]: Seems like a comment about what network we are selecting here, and why it's good enough for testing but probably not production, could be useful?

File: examples/spark/test.bats [hunk 2]
```diff
@@ -63,7 +57,8 @@ SPARK_LOCAL_DIRS=/tmp/spark
 SPARK_LOG_DIR=$spark_log
 SPARK_WORKER_DIR=/tmp/spark
 SPARK_LOCAL_IP=127.0.0.1
-SPARK_MASTER_HOST=${master_ip}
+SPARK_MASTER_HOST=${master_host}
+JAVA_HOME=/usr/lib/jvm/default-java/
```
Comment (reidpr) [inline examples/spark/test.bats:61]: Does it work to unset `JAVA_HOME`? I thought Java was smart about that if the variable wasn't already set. I.e., is the problem bogus `JAVA_HOME` leaking in from somewhere?

**Review by reidpr — COMMENTED** [review]
Could questions. Thanks for dealing with this!

File: examples/spark/test.bats (hunk 1 above)
Comment (heasterday) [reply examples/spark/test.bats:34]: I agree, I will add this.

File: examples/spark/test.bats (hunk 2 above)
Comment (heasterday) [reply examples/spark/test.bats:61]: I initially thought `JAVA_HOME` was getting incorrectly set somewhere and bypassing our `unset` as we have had issues with this in the past but that doesn't appear to be the case here. There were a few examples online of adding this `JAVA_HOME` line to the spark env file for situations where it isn't getting picked up so I took this approach.

Comment (reidpr) [reply examples/spark/test.bats:61]: Sounds good. Maybe add a comment to that effect, to head off ignorant commenters like me?

**Comment by heasterday** [comment]
Incorporated feedback, please re-review.

**Review by j-ogas — APPROVED** [review]

**Pull request merged**